### PR TITLE
Publish scaladoc to gh-pages (close #40)

### LINKED ISSUE #40: Publish scaladoc
Scaladoc can be published through 3 plugins:

- sbt-unidoc
- sbt-site
- sbt-ghpages

### PR DESCRIPTION
**Pull request by three**
Running `sbt ghpagesPushSite` will generate scaladocs and push them to the gh-pages branch.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -20,7 +23,11 @@ lazy val root = project.in(file("."))
     scalaVersion       :=  "2.11.12",
     crossScalaVersions :=  Seq("2.11.12", "2.12.5"),
     scalacOptions      :=  BuildSettings.compilerOptions,
-    javacOptions       :=  BuildSettings.javaCompilerOptions
+    javacOptions       :=  BuildSettings.javaCompilerOptions,
+
+    siteSubdirName in ScalaUnidoc := "latest/api",
+    addMappingsToSiteDir(mappings in (ScalaUnidoc, packageDoc), siteSubdirName in ScalaUnidoc),
+    gitRemoteRepo := "git@github.com:snowplow/scala-maxmind-iplookups.git"
```
Comment (BenFradet) [inline build.sbt]: I think it'd be worth it to extract those settings in their own `docSettings` variable and store them in `BuildSettings`, also you can have the remote as https: `https://github.com/snowplow/scala-maxmind-iplookups.git`

**Comment by three** [comment]
The last commit moves the settings to its own variable and changes the url. I'm still trying to figure out how everything in sbt works, so it took some time to get all the autoimports into buildsettings.

Everything seems to work by testing it on my own fork.

File: project/BuildSettings.scala [hunk 2]
```diff
@@ -60,4 +65,10 @@ object BuildSettings {
         </developer>
       </developers>)
   )
+
+  lazy val docSettings = Seq(
+    addMappingsToSiteDir(mappings in (ScalaUnidoc, packageDoc), siteSubdirName in ScalaUnidoc),
+    gitRemoteRepo := "https://github.com/snowplow/scala-maxmind-iplookups.git",
+    siteSubdirName := "."
```
Comment (BenFradet) [inline project/BuildSettings.scala]: where does `"."` publish to? aren't we fine with `""`?

Comment (three) [reply project/BuildSettings.scala]: They both publish to the toplevel directory. I can change it to `""`, I didn't think to try an empty string when I committed.

Comment (BenFradet) [reply project/BuildSettings.scala]: ok then let's just do `""` :+1:

**Comment by three** [comment]
I rebased the branch so it can merge.

**Pull request merged**